.gitlab/docker/Dockerfile.spack_build

```diff
@@ -0,0 +1,17 @@
+ARG BASE_IMAGE
+FROM $BASE_IMAGE
+
+ARG SOURCE_DIR=/pika/source
+ARG BUILD_DIR=/pika/build
+
+ARG CMAKE_COMMON_FLAGS
+ARG CMAKE_FLAGS
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+COPY . ${SOURCE_DIR}
+
+SHELL ["/bin/bash", "-c"]
+
+# Configure & Build
+RUN spack build-env $spack_spec -- bash -c "cmake -B${BUILD_DIR} ${SOURCE_DIR} $CMAKE_COMMON_FLAGS $CMAKE_FLAGS && cmake --build ${BUILD_DIR} --target tests"
```

Comment:
Could you please also explicitly build the `examples` or `all` targets? There are a few examples that are not enabled as tests, and they would not get build with just `tests`.